doc/releases/migration-guide-4.1.rst

```diff
@@ -369,6 +369,11 @@ Bluetooth Mesh
   set as deprecated. Default option for platforms that do not support TF-M
   is :kconfig:option:`CONFIG_BT_MESH_USES_MBEDTLS_PSA`.
 
+* Mesh explicitly depends on the Secure Storage subsystem if storing into
+  non-volatile memory (:kconfig:option:`CONFIG_BT_SETTINGS`) is enabled and
+  Mbed TLS library (:kconfig:option:`CONFIG_BT_MESH_USES_MBEDTLS_PSA`) is used.
+  Applications should be built with :kconfig:option:`CONFIG_SECURE_STORAGE` enabled.
```

Comment:
What if applications are built without `CONFIG_SECURE_STORAGE` enabled? If they get compile error, then perhaps, we should just add that option in existing zephyr mesh samples (and mesh_shell) code now?

Reply:
Yes, if an application is built without `CONFIG_SECURE_STORAGE` it will get compilation error due to not satisfied dependency.
It is not possible to add selection `CONFIG_SECURE_STORAGE` in mesh due to circular KConfig dependencies on SETTINGS, NVS and FLASH.